pkg/scalers/azure_data_explorer_scaler.go

```diff
@@ -159,18 +160,22 @@ func parseAzureDataExplorerAuthParams(config *ScalerConfig, logger logr.Logger)
 		}
 		metadata.ClientID = clientID
 
-		// FIXME: DEPRECATED to be removed in v2.13
-		// We should get the secret only from AuthConfig or env
-		clientSecret, err := getParameterFromConfig(config, "clientSecret", true)
+		clientSecretEnVar, err := getParameterFromConfig(config, "clientSecretFromEnv", true)
 		if err != nil {
 			return nil, err
 		}
-		if val, ok := config.TriggerMetadata["clientSecret"]; ok && val != "" {
-			logger.Info("getting 'clientSecret' from metadata is deprecated, use 'clientSecretFromEnv' or TriggerAuthentication instead")
+		clientSecretFromEnv := os.Getenv(clientSecretEnVar)
+
+		var clientSecretFromTrigger string
+		if val, ok := config.TriggerMetadata["clientSecret"]; ok {
+			clientSecretFromTrigger = val
 		}
-		// FIXME: DEPRECATED to be removed in v2.13
 
-		metadata.ClientSecret = clientSecret
+		if clientSecretFromTrigger != "" {
+			metadata.ClientSecret = clientSecretFromTrigger
+		} else if clientSecretFromEnv != "" {
+			metadata.ClientSecret = clientSecretFromEnv
+		}
```

Comment:
I think this is wrong (most probably because my comment in the code is terrible :( )
Scalers don't read envs from KEDA pod as general rule (except for really specific things), when I wrote from env, I meant from workload's envs. `getParameterFromConfig` reads the value from any source and as `clientSecret` is a secret, it has to be read only from `AuthParams `or `ResolvedEnv`